editor/src/clj/editor/gui.clj

```diff
@@ -1407,10 +1407,11 @@
 
   (property template TemplateData
             (dynamic read-only? override-node?)
-            (dynamic edit-type (g/constantly {:type resource/Resource
-                                              :ext "gui"
-                                              :to-type (fn [v] (:resource v))
-                                              :from-type (fn [r] {:resource r :overrides {}})}))
+            (dynamic edit-type (g/fnk [_node-id] {:type resource/Resource
+                                                  :ext "gui"
+                                                  :dialog-accept-fn (fn [r] (not= r (g/node-value (node->gui-scene _node-id) :resource)))
```

Comment:
This is the most frequent case.  
In the future, as part of the fix for [this issue](https://github.com/defold/defold/issues/6568), we can modify this function to filter out all the GUI files above the current GUI node in the graph (in case it's a template inside a template inside a template).